pkg/tsdb/elasticsearch/client/client_test.go

```diff
@@ -11,310 +11,276 @@ import (
+		assert.Equal(t, 56, c.GetVersion())
+	})
+
+	t.Run("When version 60 should return v6.0 client", func(t *testing.T) {
+		ds := &models.DataSource{
+			JsonData: simplejson.NewFromAny(map[string]interface{}{
+				"esVersion": 60,
+				"timeField": "@timestamp",
+			}),
+		}
+
+		c, err := NewClient(context.Background(), ds, nil)
+		require.NoError(t, err)
+		assert.Equal(t, 60, c.GetVersion())
+	})
+
+	t.Run("When version 70 should return v7.0 client", func(t *testing.T) {
+		ds := &models.DataSource{
+			JsonData: simplejson.NewFromAny(map[string]interface{}{
+				"esVersion": 70,
+				"timeField": "@timestamp",
+			}),
+		}
+
+		c, err := NewClient(context.Background(), ds, nil)
+		require.NoError(t, err)
+		assert.Equal(t, 70, c.GetVersion())
+	})
+}
+
+func TestClient_ExecuteMultisearch(t *testing.T) {
+	httpClientScenario(t, "Given a fake http client and a v2.x client with response", &models.DataSource{
+		Database: "[metrics-]YYYY.MM.DD",
+		JsonData: simplejson.NewFromAny(map[string]interface{}{
+			"esVersion": 2,
+			"timeField": "@timestamp",
+			"interval":  "Daily",
+		}),
+	}, func(sc *scenarioContext) {
+		sc.responseBody = `{
 				"responses": [
 					{
-						"hits": {	"hits": [], "max_score": 0,	"total": 4656	},
+						"hits": { "hits": [], "max_score": 0, "total": 4656 },
 						"status": 200
 					}
 				]
 			}`
 
-			Convey("When executing multi search", func() {
-				ms, err := createMultisearchForTest(sc.client)
-				So(err, ShouldBeNil)
-				res, err := sc.client.ExecuteMultisearch(ms)
-				So(err, ShouldBeNil)
+		ms, err := createMultisearchForTest(t, sc.client)
+		require.NoError(t, err)
+		res, err := sc.client.ExecuteMultisearch(ms)
+		require.NoError(t, err)
 
-				Convey("Should send correct request and payload", func() {
-					So(sc.request, ShouldNotBeNil)
-					So(sc.request.Method, ShouldEqual, http.MethodPost)
-					So(sc.request.URL.Path, ShouldEqual, "/_msearch")
+		require.NotNil(t, sc.request)
+		assert.Equal(t, http.MethodPost, sc.request.Method)
+		assert.Equal(t, "/_msearch", sc.request.URL.Path)
 
-					So(sc.requestBody, ShouldNotBeNil)
+		require.NotNil(t, sc.requestBody)
+		headerBytes, err := sc.requestBody.ReadBytes('\n')
+		require.NoError(t, err)
+		bodyBytes := sc.requestBody.Bytes()
 
-					headerBytes, err := sc.requestBody.ReadBytes('\n')
-					So(err, ShouldBeNil)
-					bodyBytes := sc.requestBody.Bytes()
+		jHeader, err := simplejson.NewJson(headerBytes)
+		require.NoError(t, err)
 
-					jHeader, err := simplejson.NewJson(headerBytes)
-					So(err, ShouldBeNil)
+		jBody, err := simplejson.NewJson(bodyBytes)
+		require.NoError(t, err)
 
-					jBody, err := simplejson.NewJson(bodyBytes)
-					So(err, ShouldBeNil)
+		assert.Equal(t, "metrics-2018.05.15", jHeader.Get("index").MustString())
+		assert.True(t, jHeader.Get("ignore_unavailable").MustBool(false))
+		assert.Equal(t, "count", jHeader.Get("search_type").MustString())
+		assert.Empty(t, jHeader.Get("max_concurrent_shard_requests"))
```

Comment:
Previously, the test did `jHeader.Get("max_concurrent_shard_requests").MustInt(10)` which meant that a default value of 10 was returned if no value was provided/set.

`max_concurrent_shard_requests` isn't set by the request, so it should be empty. Let's ensure we assert that we got an empty value.